### PR TITLE
DPKG log parsing

### PR DESCRIPTION
**Pull request by Kidswiss**
This commit contains log parsing for the DPKG log. It will parse the
contents and push it to a prometheus push gateway.

### CODE REVIEW

File: update_exporter_deb.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                        help='url for prometheus push gateway')
+
+    args = parser.parse_args()
+
+    if args.url == '':
+        print('No push gateway url given, skipping...')
+        exit()
+
+    with open('/var/log/dpkg.log', 'r') as f:
+        lines = f.readlines()
```
Comment (simu) [inline update_exporter_deb.py]: Consider restructuring finding the relevant lines of the log file to something like

```
today = date.today().strftime("%Y-%m-%d")
installed = []
for line in f:
    if 'upgrade' in line and today in line:
        installed.append(line)
```

That should keep the memory usage of the script low if the dpkg log grows big for some reason.

File: update_exporter_deb.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        g = Gauge('suc_package_upgraded', 'Information about upgraded package', [
+                  'package', 'version', 'type', 'instance'], registry=registry)
+
+        count = 0
+        for line in installed:
+            splitted_line = line.split(' ')
+            package = splitted_line[3]
+            version = splitted_line[5].strip()
+            g.labels(package, version, 'apt', socket.getfqdn()).set(1)
+            push_to_gateway(args.url, job='suc', registry=registry)
```
Comment (simu) [inline update_exporter_deb.py]: Wouldn't it make more sense to have a single push to the gateway after collecting the installed packages?

Also this loop could be merged with the loop I suggested above, ending up with code that iterates over the dpkg log exactly once.

**Review by simu — CHANGES_REQUESTED** [review]
The overall approach looks good

File: update_exporter_deb.py (hunk 1 above)
Comment (Kidswiss) [reply update_exporter_deb.py]: If I re-use this loop for building the metrics, there's no actual need for an additional array.

Comment (simu) [reply update_exporter_deb.py]: well yeah, I realized that the additional array is unnecessary after writing this comment :)

**Review by simu — APPROVED** [review]
LGTM now

**Review by simu — APPROVED** [review]

**Pull request merged**